packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -94,13 +94,16 @@ export default createRule<Options, MessageIds>({
       if (!valueType) {
         return;
       }
-
+      const value = sourceCode.getText(valueType.typeAnnotation);
+      const name = node.parent?.id?.name;
+      if (value.includes(name)) {
+        return;
+      }
```

Comment:
Hello! Thank you for the PR for my issue. I do not have experience in coding linters, but it seems to me that this simple `includes` string check could give a false positive for something like `type Tree = Record<string, AnotherTree>`, couldn't it?

Reply:
Yes, true. Thanks for letting me know. I've fixed that in the following commit.